pkg/rpc/response/errors.go

```diff
@@ -93,3 +93,11 @@ func NewSubmitError(code int64, message string) *Error {
 func (e *Error) Error() string {
 	return fmt.Sprintf("%s (%d) - %s - %s", e.Message, e.Code, e.Data, e.Cause)
 }
+
+// ExtendErrorWithData adds specified data to the error's data and cause. It returns
+// an updated error for usage simplicity.
+func ExtendErrorWithData(e *Error, data error) *Error {
```

Comment:
You're modifying global `response.ErrValidationFailed` with this and it's just not appropriate. Some `WrapErrorWithData` that will _copy_ the original value is needed here.

Reply:
Wow, I missed that. Fixed.